### PR TITLE
parallel: Allocate one channel for transferring big files

### PR DESCRIPTION
**Pull request by vadmeste**
Big files consume a lot of memory and they are also able to consume an
important bandwidth. Allocate one channel for them to be processed
separately from small files.

Working but the idea needs to be approved.

### CODE REVIEW

File: cmd/cp-main.go [hunk 1]
```diff
@@ -533,11 +533,11 @@ func doCopySession(ctx context.Context, cancelCopy context.CancelFunc, cli *cli.
 				if isCopied != nil && isCopied(cpURLs.SourceContent.URL.String()) {
 					parallel.queueTask(func() URLs {
 						return doCopyFake(ctx, cpURLs, pg, isMvCmd)
-					})
+					}, false)
 				} else {
 					parallel.queueTask(func() URLs {
 						return doCopy(ctx, cpURLs, pg, encKeyDB, isMvCmd, preserve)
-					})
+					}, cpURLs.SourceContent.Size > 128*1024*1024)
```
Comment (klauspost) [inline cmd/cp-main.go:540]: Move size to a const.

File: cmd/parallel-manager.go [hunk 2]
```diff
@@ -58,8 +58,8 @@ type ParallelManager struct {
 	// Current threads number
 	workersNum uint32
 
-	// Channel to receive tasks to run
-	queueCh chan task
+	// Channels to receive tasks to run
+	queueChannels []chan task
```
Comment (klauspost) [inline cmd/parallel-manager.go:62]: This is always 2, correct?

Either just have 2 vars or `queueChannels [2]chan task`. I prefer the former, makes everything clearer and you don't have to pass around 0 and 1, but maybe a bool.

**Review by klauspost — COMMENTED** [review]
Drive-by review :)

**Comment by vadmeste** [comment]
This can be closed, there is another PR which monitors memory consumption. Besides part size is changed to 16Mb

**Pull request closed without merging**